### PR TITLE
Add sklearn random forest as regression model

### PR DESCRIPTION
**Pull request by amueller**
The main purpose of this is to allow exporting the surrogate model to ONNX in an easy way.

The tests are copied from the HomogeneousRandomForest

### CODE REVIEW

File: source/Mlos.Python/mlos/Optimizers/RegressionModels/SklearnRandomForestRegressionModel.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+            dataframe_index=valid_rows_index
+        )
+        prediction_dataframe = predictions.get_dataframe()
+
+        if valid_rows_index is not None and not valid_rows_index.empty:
+            prediction_dataframe[is_valid_input_col] = True
+
+            predictions_array = self.random_forest_regressor_.predict(features_df)
+
+            predicted_std = _return_std(features_df, self.random_forest_regressor_.estimators_, predictions_array, min_variance=0.01)
```
Comment (bpkroth) [inline source/Mlos.Python/mlos/Optimizers/RegressionModels/SklearnRandomForestRegressionModel.py:207]: Should the `min_variance=0.01` part be configurable?

File: source/Mlos.Python/mlos/Optimizers/RegressionModels/SklearnRandomForestRegressionModel.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+        self.input_dimension_names = [dimension.name for dimension in self.input_space.dimensions]
+        self._projected_input_dimension_names = [dimension.name for dimension in self.one_hot_encoder_adapter.dimensions]
+        self.continuous_dimension_names = [dimension.name for dimension in self.one_hot_encoder_adapter.target.dimensions
+                                           if isinstance(dimension, ContinuousDimension)]
+        self.output_dimension_names = [dimension.name for dimension in self.output_space.dimensions]
+
+        self.random_forest_regressor_ = None
+
+        self.fit_X_ = None
```
Comment (bpkroth) [inline source/Mlos.Python/mlos/Optimizers/RegressionModels/SklearnRandomForestRegressionModel.py:122]: Unused?

Comment (bpkroth) [reply source/Mlos.Python/mlos/Optimizers/RegressionModels/SklearnRandomForestRegressionModel.py:122]: Actually a couple of these look like that?  Or maybe they're elsewhere in the hierarchy and I'm just misremembering.

File: source/Mlos.Python/mlos/Optimizers/BayesianOptimizerConfigStore.py [hunk 3]
```diff
@@ -35,6 +39,14 @@
                 MultiObjectiveHomogeneousRandomForest.__name__
             ])
     ).join(
+        subgrid=SklearnRandomForestRegressionModelConfig.CONFIG_SPACE,
```
Comment (byte-sculptor) [inline source/Mlos.Python/mlos/Optimizers/BayesianOptimizerConfigStore.py:42]: Could we rename CONFIG_SPACE -> parameter_space to make it consistent with other models?

File: source/Mlos.Python/mlos/Optimizers/RegressionModels/SklearnRandomForestRegressionModel.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+            input_space=input_space,
+            output_space=output_space
+        )
+
+        self.model_config = model_config
+
+        # one hot encode categorical input dimensions
+        self.one_hot_encoder_adapter = CategoricalToOneHotEncodedHypergridAdapter(
+            adaptee=self.input_space,
+            merge_all_categorical_dimensions=False)
```
Comment (byte-sculptor) [inline source/Mlos.Python/mlos/Optimizers/RegressionModels/SklearnRandomForestRegressionModel.py:112]: nit: closing parenthesis on a new line

File: source/Mlos.Python/mlos/Optimizers/RegressionModels/SklearnRandomForestRegressionModel.py [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+        self.input_dimension_names = [dimension.name for dimension in self.input_space.dimensions]
+        self._projected_input_dimension_names = [dimension.name for dimension in self.one_hot_encoder_adapter.dimensions]
+        self.continuous_dimension_names = [dimension.name for dimension in self.one_hot_encoder_adapter.target.dimensions
```
Comment (byte-sculptor) [inline source/Mlos.Python/mlos/Optimizers/RegressionModels/SklearnRandomForestRegressionModel.py:116]: is `self.continuous_dimension_names` used somewhere?

**Review by byte-sculptor — APPROVED** [review]
LTGM

**Pull request closed without merging**